src/java.base/share/classes/jdk/internal/foreign/layout/AbstractLayout.java

```diff
@@ -140,4 +150,72 @@ private static long requirePowerOfTwoAndGreaterOrEqualToOne(long value) {
         return value;
     }
 
+    public long scale(long offset, long index) {
```

Comment:
Could you add `@Override` to these as well? I think it makes it clearer that these methods might be accessed through a super type.

Reply:
Would be nice - but note that `AbstractLayoutImpl` does not extend `MemoryLayout`.